Automattic/wp-calypso
# WP-API: Move to new URL structure

We're in the process of migrating WP-API URLs as follows (ref: pMz3w-5l5-p2)

```diff
- https://public-api.wordpress.com/wp-json/sites/4/wp/v2/types
+ https://public-api.wordpress.com/wp/v2/sites/4/types

- https://public-api.wordpress.com/wp-json/sites/4/wpcom/v2/post-counts
+ https://public-api.wordpress.com/wpcom/v2/sites/4/post-counts
```

This PR incorporates previous releases of client libraries:

- https://github.com/Automattic/wpcom-xhr-request/commits/v0.5.0
- https://github.com/Automattic/wpcom.js/commits/4.9.0
- https://github.com/Automattic/wpcom-unpublished/commits/1.1.0

At the end of this chain of npm dependencies is [this change](https://github.com/Automattic/wpcom-xhr-request/commit/4e1000b53d05d9a5042d298652559bb481043850):

```diff
  var basePath = '/rest/v' + apiVersion;
  
- // if this is a wp-api request, adjust basePath
- if ( apiNamespace && wpApiNamespaces.indexOf( apiNamespace ) !== -1 ) {
+ // If this is a WP-API request, adjust basePath
+ if ( apiNamespace && /\//.test( apiNamespace ) ) {
+     // New-style WP-API URL: /wpcom/v2/sites/%s/post-counts
+     basePath = '/' + apiNamespace;
+ } else if ( apiNamespace ) {
+     // Old-style WP-API URL (deprecated): /wp-json/sites/%s/wpcom/v2/post-counts
      basePath = '/wp-json';
  }
```

If the `apiNamespace` parameter uses a slash, direct the request to the new URLs.  This PR updates Calypso to use the new code and pass namespaces with slashes (`wpcom/v2` so far).  This fits with the [intent of WP-API namespaces](http://v2.wp-api.org/extending/adding/#namespacing).

### To test

- Open the Network tab in the console and filter on `wpcom/v2`
- Visit the [post editor](http://calypso.localhost:3000/post) for a WordPress.com (not Jetpack) site that has one or more drafts, and verify that the drafts button loads a count and becomes enabled:
![image](https://cloud.githubusercontent.com/assets/227022/14122296/c61b10de-f5c0-11e5-9364-9b4ac6c8fed0.png)
- Verify that there is a successful network request to `/wpcom/v2/sites/%s/post-counts/post` (no `wp-json` prefix; `wpcom/v2` comes before `sites/%s`)
- Visit the [site settings page](http://calypso.localhost:3000/settings/general) for a WordPress.com site and verify that the "Site Timezone" dropdown loads a bunch of options from `/wpcom/v2/timezones` (no `wp-json` prefix)
